czkawka_core/src/big_file.rs

```diff
@@ -185,64 +164,53 @@ impl BigFile {
 
                     // Checking allowed extensions
                     if !self.allowed_extensions.file_extensions.is_empty() {
-                        have_valid_extension = false;
                         for extension in &self.allowed_extensions.file_extensions {
                             if file_name_lowercase.ends_with((".".to_string() + extension.to_lowercase().as_str()).as_str()) {
-                                have_valid_extension = true;
                                 break;
                             }
                         }
-                    } else {
-                        have_valid_extension = true;
+                        // Not an allowed extension, ignore it.
+                        self.information.number_of_ignored_files += 1;
+                        continue 'dir;
```

Comment:
Maybe I'm reading this bad, but if file have allowed extension, then always
```
self.information.number_of_ignored_files += 1;
                        continue 'dir;
```
is executed, but it shouldn't

Reply:
Also this seems to caused bug with temporary file finder

Reply:
This is probably the bug. The break should break out of the if statement as well.